src/test/bare-module-worker_test.ts

```diff
@@ -4,119 +4,872 @@
             },
             start: {
-              character: 6,
+              character: 8,
               line: 0,
             },
           },
         },
         filename: 'index.js',
         kind: 'diagnostic',
       },
+    ];
+    await checkTransform(files, expected, {}, cdn);
+  });
+
+  test('invalid package.json', async () => {
+    const files: SampleFile[] = [
+      {
+        name: 'index.js',
+        content: 'import "foo/index.js";',
+      },
+      {
+        name: 'package.json',
+        content: `{
+           "dependencies": XXX
+         }`,
+      },
+    ];
+    const cdn: CdnData = {
+      foo: {
+        versions: {
+          '1.0.0': {
+            files: {
+              'index.js': {
+                content: 'foo1',
+              },
+            },
+          },
+        },
+      },
+    };
+    const expected: BuildOutput[] = [
+      {
+        kind: 'file',
+        file: {
+          name: 'index.js',
+          content: 'import "./node_modules/foo@1.0.0/index.js";',
+        },
+      },
+      {
+        kind: 'file',
+        file: {
+          name: 'node_modules/foo@1.0.0/index.js',
+          content: 'foo1',
+          contentType: 'text/javascript; charset=utf-8',
+        },
+      },
+      {
+        kind: 'file',
+        file: {
+          name: 'package.json',
+          content: `{
+           "dependencies": XXX
+         }`,
+        },
+      },
+      {
+        diagnostic: {
+          message:
+            'Invalid package.json: ' +
+            {
+              chrome: 'SyntaxError: Unexpected token X in JSON at position 29',
+              firefox:
+                'SyntaxError: JSON.parse: unexpected character at line 2 column 28 of the JSON data',
+              safari:
+                'SyntaxError: JSON Parse error: Unexpected identifier "XXX"',
+            }[browser],
+          range: {
+            start:
+              browser === 'safari'
+                ? {line: 0, character: 0}
+                : {line: 1, character: 27},
+            end: {line: 2, character: 10},
+          },
+        },
+        filename: 'package.json',
+        kind: 'diagnostic',
+      },
+    ];
+    await checkTransform(files, expected, {}, cdn);
+  });
+
+  test('use import map', async () => {
+    const files: SampleFile[] = [
+      {
+        name: 'index.js',
+        content: 'import "foo";',
+      },
+    ];
+    const importMap: ModuleImportMap = {
+      imports: {
+        foo: 'http://example.com/foo',
+      },
+    };
+    const cdn: CdnData = {
+      foo: {
+        versions: {
+          '1.0.0': {
+            files: {
+              'index.js': {
+                content: 'foo',
+              },
+            },
+          },
+        },
+      },
+    };
+    const expected: BuildOutput[] = [
       {
         kind: 'file',
         file: {
           name: 'index.js',
-          content: 'import"',
+          content: 'import "http://example.com/foo";',
```

Comment:
Clarification - is there anything else we do with dynamic imports or just swap out the import string?

Could this be a relative url `/node_modules/foo.js`?

Reply:
> Clarification - is there anything else we do with dynamic imports or just swap out the import string?

Oh, I forgot to include any tests for dynamic import! Updated a test to include some (near the top of this file).

> Could this be a relative url `/node_modules/foo.js`?

This test is specifically checking the behavior of import maps, so we're not expecting the `node_modules` path at all here.

For the non-import-map cases though, we don't want use a root-relative URL like `/node_modules` (as opposed to `./node_modules`) because we don't know the root-relative path to the service-worker scope, since that is determined by the `sandboxBaseUrl` parameter and is customizable. E.g. by default we need to resolve to `https://unpkg.com/playground-elements@<version>/playground-projects/node_modules/...`, but on lit.dev it's `https://playground.lit.dev/node_modules/...`. We could pipe that parameter through to generate root-relative URLs, but it's simpler to compute paths that are file-relative.